crates/holochain/src/core/workflow/sys_validation_workflow/unit_tests.rs

```diff
@@ -0,0 +1,393 @@
+    create_action.author = previous_action.action().author().clone();
+    create_action.action_seq = previous_action.action().action_seq() + 1;
+    create_action.prev_action = previous_action.as_hash().clone();
+    create_action.timestamp = Timestamp::now().into();
+    let op = DhtOp::RegisterAgentActivity(fixt!(Signature), Action::Create(create_action));
+
+    let op_hash = test_case
+        .save_op_to_db(test_case.dht_db_handle(), op)
+        .await
+        .unwrap();
+
+    let mut network = MockHolochainP2pDnaT::new();
+    let mut ops: WireRecordOps = WireRecordOps::new();
+    ops.action = Some(Judged::valid(previous_action.clone().into()));
+    let response = WireOps::Record(ops);
+    network
+        .expect_get()
+        .return_once(move |_, _| Ok(vec![response]));
+
+    test_case.with_network_behaviour(network).run().await;
+
+    let ops_to_app_validate = test_case.get_ops_pending_app_validation().await;
+    assert!(ops_to_app_validate.contains(&op_hash));
+
+    test_case.expect_app_validation_triggered().await;
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn validate_op_with_dependency_not_found_on_the_dht() {
+    holochain_trace::test_run().unwrap();
+
+    let mut test_case = TestCase::new().await;
+
+    // Previous op, to be referenced but not found on the dht
+    let mut validation_package_action = fixt!(AgentValidationPkg);
+    validation_package_action.author = test_case.agent.clone();
+    validation_package_action.action_seq = 10;
+    let previous_action = test_case
+        .sign_action(Action::AgentValidationPkg(
+            validation_package_action.clone(),
+        ))
+        .await;
+
+    // Op to validate, to go in the dht database
+    let mut create_action = fixt!(Create);
+    create_action.author = previous_action.action().author().clone();
+    create_action.action_seq = previous_action.action().action_seq() + 1;
+    create_action.prev_action = previous_action.as_hash().clone();
+    create_action.timestamp = Timestamp::now().into();
+    let op = DhtOp::RegisterAgentActivity(fixt!(Signature), Action::Create(create_action));
+
+    test_case
+        .save_op_to_db(test_case.dht_db_handle(), op)
+        .await
+        .unwrap();
+
+    let mut network = MockHolochainP2pDnaT::new();
+    // Just return an empty response, nothing found for the request
+    let response = WireOps::Record(WireRecordOps::new());
+    network
+        .expect_get()
+        .return_once(move |_, _| Ok(vec![response]));
+
+    test_case.with_network_behaviour(network).run().await;
+
+    let ops_to_app_validate = test_case.get_ops_pending_app_validation().await;
+    assert!(ops_to_app_validate.is_empty());
+
+    // TODO Why trigger app validation if no new work was done by sys validation? App validation might need to be triggered for another reason
+    //      but is there a good reason for sys validation to just kick it off?
+    // test_case.expect_app_validation_not_triggered().await;
```

Comment:
It shouldn't trigger app validation, that seems like a mistake. Even though we are wanting app validation to get triggered more -- if we are doing it right, then this should not lead to a trigger. I'd say feel free to do the negative assertion and fix that.

Reply:
Great, yes I agree. The information about what work was done is currently not available where it would need to be to make this decision but it definitely can be. I'll get on and address this in the next PR